paloma_sdk/client/lcd/api/cw721.py

```diff
@@ -0,0 +1,424 @@
+from typing import Optional
+
+from ._base import BaseAsyncAPI, sync_bind
+from ..wallet import Wallet
+from .tx import CreateTxOptions
+from paloma_sdk.core.wasm import MsgInstantiateContract, MsgExecuteContract
+from paloma_sdk.core.coins import Coins
+from paloma_sdk.core.broadcast import BlockTxBroadcastResult
+__all__ = ["AsyncCw721API", "Cw721API"]
+
+class AsyncCw721API(BaseAsyncAPI):
+    async def instantiate(
+        self,
+        wallet: Wallet,
+        code_id: int,
+        name: str,
+        symbol: str,
+        minter: str
+    ) -> BlockTxBroadcastResult:
+        """instantiate the Cw721 smart contract using code id.
+        Args:
+            wallet (Wallet): CW721 deployer wallet
+            code_id (int): Code_id of CW721 code
+            name (str): CW721 token name
+            symbol (str): CW721 token symbol
+            minter (str): CW721 token minter
+        Returns:
+            BlockTxBroadcastResult: Transaction Broadcast Result
+        """
+        instantiate_msg = {
+            "name": name,
+            "symbol": symbol,
+            "minter": minter
+        }
+        funds = Coins()
+        tx = await wallet.create_and_sign_tx(CreateTxOptions(
+            msgs=[MsgInstantiateContract(
+                wallet.key.acc_address,
+                None,
+                code_id,
+                "CW721",
+                instantiate_msg,
+                funds
+            )]
+        ))
+        result = await self._c.tx.broadcast(tx)
+        return result
+
+    async def mint(
+        self,
+        wallet: Wallet,
+        token: str,
+        token_id: str,
+        owner: str,
+        token_uri: str
+    ) -> BlockTxBroadcastResult:
+        """Mint CW721 token
+        Args:
+            wallet (Wallet): CW721 sender wallet
+            token (str): CW721 token address
+            token_id (str): CW721 token id
+            owner (str): owner who will receive the CW721
+            token_uri (str): URI of the minting token
+        Returns:
+            BlockTxBroadcastResult: Transaction Broadcast Result
+        """
+        execute_msg = {"mint": {
+                "token_id": token_id,
+                "owner": owner,
+                "token_uri": token_uri,
+                "extension": None
+            }}
+
+        funds = Coins()
+        tx = await wallet.create_and_sign_tx(CreateTxOptions(
+            msgs=[MsgExecuteContract(
+                wallet.key.acc_address,
+                token,
+                execute_msg,
+                funds
+            )]
+        ))
+        result = await self._c.tx.broadcast(tx)
+        return result
+
+    async def approve(
+        self,
+        wallet: Wallet,
+        token: str,
+        spender: str,

```

Comment:
Why is this a type  `string` and not a `wallet`?

Reply:
`token` and `spender` are string types. We don't have private keys for the `token` and the `spender`.